### DISCUSSION TITLE
MacOS 客户端问题

### DISCUSSION THREAD

**Discussion by iveney** [1715173329/UnblockNeteaseMusic#715] [Q&A] [answered]
最近从一部旧的 Macbook Pro 换到新的 Macbook Pro (M1)，发现客户端 (2.3.2) 用不了了，打开后会一直显示播放失败，网络错误，所有歌曲都失败并且前进到下一首

![image](https://user-images.githubusercontent.com/51033/170604934-27686de3-a73f-4a02-a30c-0d2d71938dcb.png)

从 log 上来看是根本没有 hit server。但是这个 2.3.2 版本在旧 MBP 是能用的。尝试着用旧版本 2.0.0，可以在新 MBP 上用。旧版可以凑合着用但是不支持 Dark Theme 看着有点不舒服，请问有人有类似的问题吗？或者能使用的最新的 Mac 版本？ 

使用旧版成功使用时的 Log：

```
➜  UnblockNeteaseMusicServer git:(enhanced) node app.js -p 80:443 -f 103.126.92.133
INFO: (app) HTTP Server running @ http://0.0.0.0:80
INFO: (app) HTTPS Server running @ http://0.0.0.0:443
INFO: (spawn) running yt-dlp -f 140 --dump-json ytsearch1:奔向世界的尽头 - 打扰一下乐团
INFO: (spawn) running yt-dlp -f 140 --dump-json https://www.youtube.com/watch?v=pxVgRXEOeLw
INFO: (provider/match) Replaced: [1473633505] 奔向世界的尽头
    audioId: 1473633505
    songName: "奔向世界的尽头"
    url: "http://freetyst.nf.migu.cn/public/product9th/product44/2021/07/1509/2021%E5%B9%B407%E6%9C%8814%E6%97%A521%E7%82%B927%E5%88%86%E7%B4%A7%E6%80%A5%E5%86%85%E5%AE%B9%E5%87%86%E5%85%A5%E9%9F%B6%E6%84%94%E9%9F%B3%E4%B9%905018%E9%A6%96/%E6%A0%87%E6%B8%85%E9%AB%98%E6%B8%85/MP3_320_16_Stero/69990100104094906.mp3"
```

**Comment by gqbre** [comment]
iOS， Mac 都不行了。应该是 api 格式变化

**Reply by iveney** [reply]
这个是您的大作？ https://www.sheyilin.com/2019/06/unblockneteasemusic

我根据上文，发现本地运行 + proxfier 还是可以的 (2.3.8)。但是放到 VPS （用的 GCP）又不行了……

## 本地
`node app.js -p 8000:8043 -s -e https://music.163.com`
proxifier 用的 HTTPS proxy 127.0.0.1:8000

## 远程
`node app.js -p 80:443 -s -e https://music.163.com`
proxifier 用的 HTTPS proxy <PUBLIC_IP>:80。表现跟 issue 原文一样。

**Comment by 1715173329** [comment, marked as answer]
请设置 endpoint，具体参见其他 discussion 和 issue。

**Reply by iveney** [reply]
参考了 https://github.com/UnblockNeteaseMusic/server/discussions/682#discussioncomment-2738214 加了 -e 后倒是没有出现上面的现象了，但同时也等于没用 UNM 一样，歌曲都是灰的。

```
➜  UnblockNeteaseMusicServer git:(enhanced) node app.js -p 80:443 -f 103.126.92.133 -e https://music.163.com
INFO: (app) HTTP Server running @ http://0.0.0.0:80
INFO: (app) HTTPS Server running @ http://0.0.0.0:443
```

我用的是本地直接改 `/etc/hosts` 不知道有没影响。
```
127.0.0.1 music.163.com
127.0.0.1 interface.music.163.com
```

**Reply by 1715173329** [reply]
信任证书了吗？

**Reply by iveney** [reply]
貌似还真是证书的问题，之前一直用的是 nondanee 的证书与代理。local & remote migrate 到 UNM-server 后，手动删除证书，重新导入 UNM-server 的 ca 并且信任，macOS 能用了，iOS 的 proxy.pac 好像还是不行。

感谢！
